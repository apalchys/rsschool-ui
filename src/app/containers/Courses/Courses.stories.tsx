import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { ReduxProviderDecorator } from '../../storybook/ReduxProviderDecorator';
import Courses from './index';

const noop = () => {};

storiesOf('Courses', module)
    .addDecorator(storyFn => <ReduxProviderDecorator>{storyFn()}</ReduxProviderDecorator>)
    .add('No Data', () => {
        return <Courses courses={[]} feed={[]} participations={[]} enrollUser={noop} fetchCourses={noop} />;
    });
