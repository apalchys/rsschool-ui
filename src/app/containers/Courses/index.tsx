import CoursesEnrollment from 'components/CoursesEnrollment';
import { enrollUserInCourse, fetchAllCourses, fetchUserParticipations } from 'core/actions';
import { ICourse, IFeedRecord, IUserParticipation } from 'core/models';
import { RootState } from 'core/reducers';
// import { classNames } from 'core/styles';
import * as React from 'react';
import { connect } from 'react-redux';

// const styles = require('./index.scss');
// const cn = classNames(styles);

const mapStateToProps = (state: RootState, props: Props): Props => {
    if (state.courses == null || state.user == null) {
        return props;
    }
    return {
        ...props,
        courses: state.courses.data || [],
        participations: state.user.participations || [],
        feed: state.user.feed || [],
    };
};

const mapDispatchToProps = (dispatch: any, props: Props): Props => {
    return {
        ...props,
        fetchCourses: () => {
            dispatch(fetchAllCourses());
            dispatch(fetchUserParticipations());
        },
        enrollUser: (id: string) => {
            dispatch(enrollUserInCourse(id));
        },
    };
};

type Props = {
    courses: ICourse[];
    enrollUser: (courseId: string) => void;
    feed: IFeedRecord[];
    fetchCourses: () => void;
    participations: IUserParticipation[];
};

class Courses extends React.Component<Props, any> {
    componentDidMount() {
        this.props.fetchCourses();
    }

    render() {
        return (
            <div>
                <CoursesEnrollment
                    courses={this.props.courses || []}
                    participations={this.props.participations || []}
                    enroll={(courseId: string) => this.props.enrollUser(courseId)}
                />
            </div>
        );
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Courses);
