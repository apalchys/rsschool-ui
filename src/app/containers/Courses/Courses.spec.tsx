import { shallow } from 'enzyme';
import * as React from 'react';
import Courses from './index';
const { __stateGetter } = require('react-redux');

const noop = () => {};

describe('Courses', () => {
    beforeEach(() => {
        __stateGetter.mockImplementation(() => ({}));
    });

    it('renders correctly', () => {
        const output = shallow(
            <Courses courses={[]} participations={[]} enrollUser={noop} fetchCourses={noop} feed={[]} />,
        );
        expect(output).toMatchSnapshot();
    });
});
