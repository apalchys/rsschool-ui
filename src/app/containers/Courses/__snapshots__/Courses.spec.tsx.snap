// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses renders correctly 1`] = `
<Courses
  courses={Array []}
  enrollUser={[Function]}
  feed={Array []}
  fetchCourses={[Function]}
  participations={Array []}
/>
`;
